reviews/graduation-nats.md

```diff
@@ -0,0 +1,35 @@
+# NATS Graduation Application
+
+NATS was accepted into the CNCF as an incubating project in March 2018. Significant progress has been made in 2018 to improve process and further mature NATS as a project under the guidance of the CNCF.
+
+## The following represents the successful CNCF graduation criterion met by NATS.
+
+### Committers
+**Have committers from at least two organizations.**
+
+NATS has committers from [various organizations](https://nats.devstats.cncf.io/d/5/companies-summary?orgId=1&var-period_name=Last%20decade&var-metric=repositories).
```

Comment:
Just to clarify, the canonical list of committers are those listed are core-maintainers in the MAINTAINERS.MD file of each project, right?, e.g.:
https://github.com/nats-io/gnatsd/blob/master/MAINTAINERS.md
https://github.com/nats-io/nats-streaming-server/blob/master/MAINTAINERS.md
https://github.com/nats-io/go-nats/blob/master/MAINTAINERS.md
https://github.com/nats-io/go-nats-streaming/blob/master/MAINTAINERS.md
https://github.com/nats-io/ruby-nats/blob/master/MAINTAINERS.md
https://github.com/nats-io/node-nats/blob/master/MAINTAINERS.md

Superficially it seems that Derek Collison, Ivan Kozlovic, Alberto Ricart and Waldemar Quevedo are the only core maintainers of all of the projects, and are all from Synadia?   Is that right?  Could you clarify?

Reply:
@quinton-hoole-2 , you are correct, most of our repositories have been contributed by Synadia employees (Apcera employees at the time) and continue to be maintained by the same people.  We do have at least one outside maintainer - the nats-operator comes to mind.  We have been and continue to be open to outside contributions and adding new maintainers.

Reply:
Apologies for not responding sooner, but we wanted to respond with progress related to this area of concern.  

While we did have two different companies represented you are correct in your assessment that the nats-operator is not a core component of NATS.  Over the last five months we've expanded the NATS team to now have three companies represented and three maintainers preferring to remain independent.  Our governance gives us the ability to grant any type of maintainer permission to commit to repositories, so maintainers with commit permission don't necessarily need to be core maintainers.

Today, alongside maintainers at Synadia we have maintainers with commit permission (merge PRs) on the following repositories:

- NATS Streaming Server (R.I. Pienaar)
- C# NATS client and C# NATS Streaming client (GE, Christopher Watford)
- Python client, asycnio client (Workiva, Charlie Strawn and Brian Shannan)
- NGINX client module (Lev Brock)
- NATS operator (Paulo Pires)

These constitute some of our more popular clients.  The NATS streaming server is core to the NATS project, and is widely used with over 30M docker hub pulls.  With the growth of the NATS team and additional maintainers with commit permissions, do we adequately address the concern the TOC has regarding the number of committers from different companies?

Reply:
Hi @ColinSullivan1 .  This is great progress.

To be clear, the primary goal here is to ensure that the ongoing development and maintenance of the project does not de-facto rely solely on continued active involvement of a single company or independent individual.  This is in turn to provide consumers of NATS technology with a good level of confidence that the project is highly likely to be around and well-maintained in the future.   This should not in any way be misinterpreted to be casting doubt on Synadia's continued strong involvement.  It's just that in general, companies can change strategies or priorities over time, be acquired, or go out of business, and consumers of graduated CNCF projects should be able to have high confidence that such an event will not materially affect the long-term sustainability of the project.

It's not yet totally clear to me that we have achieved that goal, given that both gnatsd and NATS Streaming Server, the heart of the system, still both appear to be very much de-facto dependent on Synadia employees active and continued involvement? 

Could you perhaps sketch out for us what might realistically happen if all Synadia employees were to (hypothetically of course) immediately cease to be involved at all, for whatever reasons.

As an example, for Kubernetes, the community and users have a high level of confidence that if any one of the say three largest contributing companies were to completely disengage, the remaining project contributors would be willing and able to continue to support and nurture the project into the future.   Is this the case for NATS?  If so, could you provide some motivation to assure users?

Many thanks....

Reply:
PS:  Please feel free to grab me for a call to discuss, if that's easier than bouncing PR comments back and forth.

Reply:
Might be good to get on a call, can you suggest some times that work for you?

derek@nats.io
colin@nats.io

Thanks.

Reply:
My apologies - I totally dropped the ball on this one.  Will pick it up again.

Reply:
I had this discussion with @derekcollison and his team on 04/22.  I think we all understand each other clearly, and the plan is to update this proposal accordingly, ASAP.